### ISSUE TITLE
Model tuning

### ISSUE THREAD

**Issue by millerh1** [dmontemayor/jdrfmetabo#16]
We need to come up with an approach for tuning the model. 

@dmontemayor Proposes that we try to tune the hyperparameters such that we maximize the sparsity -- potentially weighted by the reconstruction error

**Comment by dmontemayor** [comment]
suggest we add an optional arg to the fit function like the validation fraction arg that scales how much the optimization weights the loss function vs the sparsity metric. maybe call it sparsity_optmization with range [0.,1.]?

**Comment by millerh1** [comment]
New subtasks:

- [x] Need to figure out whether the sparsity is being measured correctly
- [x] Need to test the sparsity of the U matrix
- [x] Need to scan a smaller range of regularization params
- [x] Need a sparsity vs performance plot
- [x] Need to come up with a "sweet spot" estimate. Need most interpretable model within the cross-over range of sparsity vs performance. 
- [ ] Can we come up with a method for setting the number of LVs? Maybe number of PCA components that explain X percent of variance -- this gives the range for LV number (future direction)
- [ ] Sweep through various numbers of LBs: 100, 200, ... For each, compute cell type classification performance -- what % variance would be explained by that number of PCs?

@Julian-Borbeck

**Comment by millerh1** [comment]
Transferred to https://github.com/Bishop-Laboratory/Cancer-Normal-Transformer-Analysis/issues/27
